src/celeritas/user/DetectorSteps.cu

```diff
@@ -117,21 +141,25 @@ void copy_steps<MemSpace::device>(
 {
     CELER_EXPECT(output);
 
-    // Copy the original detector IDs
-    thrust::device_vector<DetectorId> orig_ids;
-    {
-        auto detector_span = state.detector[AllItems<DetectorId>{}];
-        orig_ids.assign(device_pointer_cast(detector_span.begin()),
-                        device_pointer_cast(detector_span.end()));
-    }
+    // Store the thread IDs of active tracks that are in a detector
+    auto start = thrust::device_pointer_cast(state.valid_id.data().get());
+    auto end = thrust::copy_if(
+        thrust::device,
+        thrust::make_counting_iterator(size_type(0)),
+        thrust::make_counting_iterator(state.size()),
```

Comment:
Is there any practical difference between thrust's counting iterators and our `range_iter`? 😉 asking just in case we eventually move to something like Alpaka...

Reply:
Nope, not that I'm aware of.